standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/client/HookMetaStoreClientProxy.java

```diff
@@ -0,0 +1,1266 @@
+    return FilterUtils.filterTableNamesIfEnabled(
+        isClientFilterEnabled, filterHook, catName, dbName, tableNames);
+  }
+
+  @Override
+  public List<String> getTables(String dbname, String tablePattern) throws MetaException {
+    try {
+      return getTables(getDefaultCatalog(conf), dbname, tablePattern);
+    } catch (Exception e) {
+      MetaStoreUtils.throwMetaException(e);
+    }
+    return null;
+  }
+
+  @Override
+  public List<String> getTables(String catName, String dbName, String tablePattern) throws TException {
+    // SG:FIXME, inefficient
+    List<String> tableNames = getDelegate().getTables(catName, dbName, tablePattern);
+    // tables is filtered by FilterHook.
+    List<Table> tables = getTableObjectsByName(catName, dbName, tableNames);
+    List<String> result = new ArrayList<>();
+    for (Table tbl: tables) {
+      result.add(tbl.getTableName());
+    }
+    return result;
+  }
+
+  @Override
+  public List<String> getTables(String dbname, String tablePattern, TableType tableType) throws MetaException {
+    try {
+      return getTables(getDefaultCatalog(conf), dbname, tablePattern, tableType);
+    } catch (Exception e) {
+      MetaStoreUtils.throwMetaException(e);
+    }
+    return null;
+  }
+
+  @Override
+  public List<String> getTables(String catName, String dbName, String tablePattern,
+      TableType tableType) throws TException {
+    List<String> tables = getDelegate().getTables(catName, dbName, tablePattern, tableType);
+    return FilterUtils.filterTableNamesIfEnabled(isClientFilterEnabled, filterHook, catName, dbName, tables);
+  }
+
+  @Override
+  public List<Table> getAllMaterializedViewObjectsForRewriting() throws TException {
+    try {
+      List<Table> views = getDelegate().getAllMaterializedViewObjectsForRewriting();
+      return FilterUtils.filterTablesIfEnabled(isClientFilterEnabled, filterHook, views);
+    } catch (Exception e) {
+      MetaStoreUtils.throwMetaException(e);
+    }
+    return null;
+  }
+
+  @Override
+  public List<String> getMaterializedViewsForRewriting(String dbName) throws TException {
+    return getMaterializedViewsForRewriting(getDefaultCatalog(conf), dbName);
+  }
+
+  @Override
+  public List<String> getMaterializedViewsForRewriting(String catName, String dbname)
+      throws MetaException {
+    try {
+      List<String> views = getDelegate().getMaterializedViewsForRewriting(catName, dbname);
+      return FilterUtils.filterTableNamesIfEnabled(isClientFilterEnabled, filterHook, catName, dbname, views);
+    } catch (Exception e) {
+      MetaStoreUtils.throwMetaException(e);
+    }
+    return null;
+  }
+
+  @Override
+  public List<TableMeta> getTableMeta(String dbPatterns, String tablePatterns, List<String> tableTypes)
+      throws MetaException {
+    try {
+      return getTableMeta(getDefaultCatalog(conf), dbPatterns, tablePatterns, tableTypes);
```

Comment:
This needs to call `getDeletage().getTableMeta(String dbPatterns, String tablePatterns, List<String> tableTypes)`